src/main.ts

```diff
@@ -355,15 +355,24 @@ export default class AdvancedURI extends Plugin {
                 dataToWrite = lines.join("\n");
             }
         } else {
-            let fileData: string;
             if (file instanceof TFile) {
-                fileData = await this.app.vault.read(file);
                 path = file.path;
+                const fileData = await this.app.vault.read(file);
+                if (parameters.line) {
+                    let line = Math.max(Number(parameters.line) - 1, 0);
+                    const lines = fileData.split("\n");
+                    if (lines[line]?.trim() !== "") {
```

Comment:
What's the reason for this check? Why are you handling an empty line differently?

Reply:
My idea for appending was that if the line is empty, it should append to that line, and if it already contains something, it should add it to the next line (which is what I tried to describe in the PR description, though it may be a little confusing). I'm not married to this behavior, so let me know if there's something else that's more intuitive.

Reply:
Okay, I understood the behavior now. I think we should handle them equally. So that we keep the empty line.